src/server/src/modules/user.ts

```diff
@@ -104,10 +104,54 @@ export const getUserByEmail = async (userId: number, email: string) => {
         message: '',
     };
 
+    const receiver = await findUserByEmail(email)
+
+    // Check if the user with the email exists.
+    if (!receiver) {
+        getUserByEmailResult.message =
+            'A user with the email does not exist.';
+        return getUserByEmailResult;
+    }
+
+    if (receiver.id == userId) {
+        getUserByEmailResult.message =
+            'You cannot send message to yourself.';
+        return getUserByEmailResult;
+    }
+
+    getUserByEmailResult.message = 'Successful';
```

Comment:
I think you can move this line after getUserByIdResult is found

Reply:
Done.